src/components/TodoMain/TodoMain.tsx

```diff
@@ -0,0 +1,100 @@
+import cn from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[] | null;
+  tempTodo: Todo | null;
+  deleteTodo: (id: number) => void;
+};
+
+export const TodoMain: React.FC<Props> = ({ todos, tempTodo, deleteTodo }) => {
+  const [isEditActive, setIsEditActive] = useState<number>(-1);
+  const [loading, setLoading] = useState(0);
+
+  const onSubmitChanges = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setIsEditActive(-1);
+  };
+
+  const onDelete = (id: number) => {
+    setLoading(id);
+    deleteTodo(id);
+  };
+
+  return (
+    <section className="todoapp__main">
+      {todos?.map((todo) => {
+        const { completed, title, id } = todo;
+
+        return (
+          <div className={cn('todo', { completed })} key={id}>
+            <label className="todo__status-label">
+              <input type="checkbox" className="todo__status" checked />
+            </label>
+
+            {isEditActive === id ? (
```

Comment:
seems weird, because I expect `isEditActive` to store a boolean value, and here it is compared with the id, which is obviously a number